<?xml version="1.0"?>
<launch>
  <!-- CARLA 서버 설정 -->
  <arg name="host" default="localhost" />
  <arg name="port" default="2000" />
  
  <!-- 차량 설정 -->
  <arg name="num_vehicles" default="10" doc="스폰할 차량 수" />
  <arg name="traffic_manager_port" default="8000" doc="Traffic Manager 포트" />
  <arg name="safe_mode" default="true" doc="안전 모드 (동기화 비활성화)" />
  
  <!-- 교통 차량 스폰 노드 -->
  <node pkg="carla_camera_stream" 
        type="spawn_traffic_vehicles.py" 
        name="carla_traffic_vehicle_spawner" 
        output="screen">
    <param name="host" value="$(arg host)" />
    <param name="port" value="$(arg port)" />
    <param name="num_vehicles" value="$(arg num_vehicles)" />
    <param name="traffic_manager_port" value="$(arg traffic_manager_port)" />
    <param name="safe_mode" value="$(arg safe_mode)" />
  </node>
</launch>

